src/analyses/goto_check.cpp

```diff
@@ -1412,6 +1412,14 @@ void goto_checkt::check_rec(
     pointer_rel_check(expr, guard);
   else if(expr.id()==ID_dereference)
     pointer_validity_check(to_dereference_expr(expr), guard);
+
+  else if(expr.id() == ID_side_effect && to_side_effect_expr(expr).get_statement() == ID_nondet)
```

Comment:
For code consistency: could you line break at && to avoid lines longer than 70 characters? Also, no whitespace around ==.